### ISSUE TITLE
Touchscreen Lightgun support for RetroArch Android & IOS please

### ISSUE THREAD

**Issue by Jonathan7299** [flyinghead/flycast#604]
Can you guys add touchscreen lightgun support for the Flycast core on RetroArch Android and IOS?

**Comment by flyinghead** [comment]
This is something that must be done in RetroArch. The Flycast core already supports lightguns.

**Comment by Jonathan7299** [comment]
I was talking about the Touchscreen Lightgun that the MAME core has (where wherever you touch on the screen is where it will fire). If it already has an option that does that, can you please tell me how to do it?

**Comment by flyinghead** [comment]
Right, it looks like this is possible but needs some support in the core itself:
https://www.libretro.com/index.php/retroarch-in-development-touchscreen-lightgun-for-iosandroid-iphone-xs-max-shown/

**Comment by Jonathan7299** [comment]
Thanks 👍

**Comment by flyinghead** [comment]
Let's keep the original issue open.

**Comment by StormedBubbles** [comment]
Hi,

I believe this can be accomplished using the `RETRO_DEVICE_POINTER` type, which can be set as a separate input device from the `RETRO_DEVICE_LIGHTGUN` type and has its own coordinate system (`RETRO_DEVICE_ID_POINTER_X` and `RETRO_DEVICE_ID_POINTER_Y`, though I think these behave the same as the lightgun coordinates) and

- `RETRO_DEVICE_ID_POINTER_PRESSED` (I think typically used to emulate the trigger)
- `RETRO_DEVICE_ID_POINTER_COUNT`, which should count the number of simultaneous touches on a screen that allows multi-touch inputs and allow you to pass that number to an emulator input. That way, you could have one finger activate the trigger, two fingers reload, etc. 

I know a little bit about the Libretro input code and can help troubleshoot some issues if needed, but I probably wouldn't be the best person to add this feature from scratch. 😆

**Comment by kireily** [comment]
> Right, it looks like this is possible but needs some support in the core itself: https://www.libretro.com/index.php/retroarch-in-development-touchscreen-lightgun-for-iosandroid-iphone-xs-max-shown/

This is soooooo coooool!! I I dream of this to play the house of dead2 of DC in Flycast ios,will it be possible?

**Comment by flyinghead** [comment]
Should be available in the next core update. Choose "Pointer" as input device. One touch to shoot, two or more touches to reload.

**Comment by pinospin** [comment]
so is it support for psvita port?
after all, it also have touchscreen
